<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" >
    <param name="camera_info_url" value="file://$(find vendor_hand)/config/apriltag/camera.yaml"/>
    <param name="image_width" value="1920"/>
    <param name="image_height" value="1080"/>
    <param name="video_device" value="/dev/video0"/>
    <param name="autofocus" value="true"/>
  </node>

  <node name="image_proc" pkg="image_proc" type="image_proc" >
    <remap from="image_raw" to="usb_cam/image_raw" />
    <remap from="camera_info" to="usb_cam/camera_info" />
  </node>

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true">
    
    <remap from="image_rect" to="image_rect_color" />
    <remap from="camera_info" to="usb_cam/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" />
    
    <rosparam command="load" file="$(find vendor_hand)/config/apriltag/settings.yaml"/>
    <rosparam command="load" file="$(find vendor_hand)/config/apriltag/tags.yaml"/>
  </node>

</launch>